execution/executor/src/components/chunk_output.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::components::apply_chunk_output::ApplyChunkOutput;
+use anyhow::Result;
+use diem_crypto::hash::TransactionAccumulatorHasher;
+use diem_state_view::StateView;
+use diem_types::{
+    proof::accumulator::InMemoryAccumulator,
+    protocol_spec::DpnProto,
+    transaction::{Transaction, TransactionOutput},
+};
+use diem_vm::VMExecutor;
+use executor_types::ExecutedChunk;
+use fail::fail_point;
+use std::{collections::HashSet, sync::Arc};
+use storage_interface::state_view::{StateCache, VerifiedStateView};
+
+pub struct ChunkOutput {
+    /// Input transactions.
+    pub transactions: Vec<Transaction>,
+    /// Raw VM output.
+    pub transaction_outputs: Vec<TransactionOutput>,
+    /// Carries the frozen base state view, so all in-mem nodes involved won't drop before the
+    /// execution result is processed; as well as al the accounts touched during execution, together
+    /// with their proofs.
+    pub state_cache: StateCache,
+}
+
+impl ChunkOutput {
+    pub fn by_transaction_execution<V: VMExecutor>(
+        transactions: Vec<Transaction>,
+        state_view: VerifiedStateView<DpnProto>,
+    ) -> Result<Self> {
+        let transaction_outputs = V::execute_block(transactions.clone(), &state_view)?;
```

Comment:
was there a timer for this before?

Reply:
no https://github.com/diem/diem/blob/2289c823fde8c9511e5d124e4e81ef1d9cf12e21/execution/executor/src/lib.rs#L531

will add the one from the BlockExecutor

Reply:
actually it's called "block execution timer". will add a separate one on the call site